include/mqtt/endpoint.hpp

```diff
@@ -12453,23 +12459,29 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
         void operator()(
             boost::system::error_code const& ec,
             std::size_t bytes_transferred) const {
-            if (func_) func_(ec);
+            func_(ec);
             for (std::size_t i = 0; i != num_of_messages_; ++i) {
                 self_->queue_.pop_front();
             }
             if (ec || // Error is handled by async_read.
                 !self_->connected_) {
                 self_->connected_ = false;
                 while (!self_->queue_.empty()) {
-                    self_->queue_.front().handler()(ec);
+                    // Handlers for outgoing packets need not be valid.
+                    if(self_->queue_.front().handler()) {
```

Comment:
```suggestion
                    if(auto const& h = self_->queue_.front().handler()) {
                        h(ec);
```

I'm not sure `const` is required or not.

Reply:
In this situation it should be `auto &&`

Reply:
I'm not at a computer (phone only) so can't edit the code myself.

You can make this change if you want. I don't mind.